weeklyChallenges04/ChallengesSet04.cs

```diff
@@ -1,52 +1,136 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 namespace weeklyChallenges04
 {
     public class ChallengesSet04
     {
         public int AddEvenSubtractOdd(int[] numbers)
         {
-            throw new NotImplementedException();
+            int numTotal = 0;
+            foreach (int num in numbers)
+            {
+                if (num % 2 == 0)
+                {
+                    numTotal += num;
+                }
+                else
+                {
+                    numTotal -= num;
+                }
+            }
+            return numTotal;
+
         }
 
         public int GetLengthOfShortestString(string str1, string str2, string str3, string str4)
         {
-            throw new NotImplementedException();
+            List<int> intList = new List<int>();
+            intList.Add(str1.Length);
+            intList.Add(str2.Length);
+            intList.Add(str3.Length);
+            intList.Add(str4.Length);
+
+            return intList.Min();
         }
 
         public int GetSmallestNumber(int number1, int number2, int number3, int number4)
         {
-            throw new NotImplementedException();
+            List<int> intList = new List<int>();
+            intList.Add(number1);
+            intList.Add(number2);
+            intList.Add(number3);
+            intList.Add(number4);
+
+            return intList.Min();
         }
 
         public void ChangeBusinessNameTo_TrueCoders(Business biz)
         {
-            throw new NotImplementedException();
+            biz.Name = "TrueCoders";
         }
 
         public bool CouldFormTriangle(int sideLength1, int sideLength2, int sideLength3)
         {
-            throw new NotImplementedException();
+            if(sideLength1 + sideLength2 > sideLength3 & sideLength1 + sideLength3 > sideLength2 & sideLength2 + sideLength3 > sideLength1)
+            {
+                return true;
+            }
+            return false;
+
         }
 
         public bool IsStringANumber(string input)
         {
-            throw new NotImplementedException();
+            double n = 0;
+            return double.TryParse(input, out n);
```

Comment:
this could've been refactored into one line, with the double variable declared within that line:
`
return double.TryParse(input, out **double** n);`